rust/chains/tw_polkadot/tests/extrinsic.rs

```diff
@@ -0,0 +1,489 @@
+use std::borrow::Cow;
+use std::default::Default;
+
+use tw_encoding::hex::ToHex;
+use tw_number::U256;
+use tw_polkadot::extrinsic::Extrinsic;
+use tw_proto::Polkadot::Proto;
+use tw_proto::Polkadot::Proto::mod_Balance::{AssetTransfer, BatchAssetTransfer, Transfer};
+use tw_proto::Polkadot::Proto::mod_Identity::mod_AddAuthorization::{AuthData, Data};
+use tw_proto::Polkadot::Proto::mod_Staking::{
+    Bond, BondExtra, Chill, Nominate, Rebond, Unbond, WithdrawUnbonded,
+};
+
+#[test]
+fn polymesh_encode_transfer_with_memo() {
+    let input = Proto::SigningInput {
+        network: 12,
+        multi_address: true,
+        message_oneof: Proto::mod_SigningInput::OneOfmessage_oneof::balance_call(Proto::Balance {
+            message_oneof: Proto::mod_Balance::OneOfmessage_oneof::transfer(Transfer {
+                to_address: "2EB7wW2fYfFskkSx2d65ivn34ewpuEjcowfJYBL79ty5FsZF".into(),
+                value: Cow::Owned(U256::from(1u64).to_little_endian().to_vec()),
+                memo: "MEMO PADDED WITH SPACES".into(),
+                call_indices: Some(Proto::CallIndices {
+                    variant: Proto::mod_CallIndices::OneOfvariant::custom(
+                        Proto::CustomCallIndices {
+                            module_index: 0x05,
+                            method_index: 0x01,
+                        },
+                    ),
+                }),
+            }),
+        }),
+        ..Default::default()
+    };
+
+    let extrinsic = Extrinsic::from_input(input);
+    let encoded = extrinsic.encode_call().expect("error encoding call");
+    assert_eq!(
+        encoded.to_hex(),
+        "0501004c6c63e3dc083959f876788716b78885460b5f3c7ed9379f8d5f408e08639e0204014d454d4f20504144444544205749544820535041434553000000000000000000"
+    );
+}
+
+#[test]
+fn polymesh_encode_authorization_join_identity() {
+    let input = Proto::SigningInput {
+        network: 12,
+        multi_address: true,
+        message_oneof: Proto::mod_SigningInput::OneOfmessage_oneof::polymesh_call(
+            Proto::PolymeshCall {
+                message_oneof: Proto::mod_PolymeshCall::OneOfmessage_oneof::identity_call(
+                    Proto::Identity {
+                        message_oneof: Proto::mod_Identity::OneOfmessage_oneof::add_authorization(
+                            Proto::mod_Identity::AddAuthorization {
+                                call_indices: Some(Proto::CallIndices {
+                                    variant: Proto::mod_CallIndices::OneOfvariant::custom(
+                                        Proto::CustomCallIndices {
+                                            module_index: 0x07,
+                                            method_index: 0x0d,
```

Comment:
Highly nested code. Could you please move some parts to functions?
For example,
```rust
fn custom_call_indices(module: u8, method: u8) -> Option<Proto::CallIndices> {
Some(Proto::CallIndices {
                                    variant: Proto::mod_CallIndices::OneOfvariant::custom(
                                        Proto::CustomCallIndices {
                                            module_index: module as u32,
                                            method_index: method as u32,
// ...
}
```